docs/sql/statements/select.rst

```diff
@@ -304,6 +304,44 @@ each group.
   column from the queried relation rather than an output column name.
 
 
+.. _sql-select-group-by-all:
+
+``GROUP BY ALL``
+''''''''''''''''
+
+``GROUP BY ALL`` is a shorthand syntax that groups all non-aggregate columns in
+the SELECT list. This eliminates the need to explicitly list all grouping
+columns.
+
+::
+
+   GROUP BY ALL [HAVING condition]
+
+When using ``GROUP BY ALL``, the following rules apply:
+- All output columns that are not aggregate functions are used as grouping columns
+- Aggregate functions (like COUNT, SUM, AVG etc.) are excluded from the automatic grouping
+- The order of columns in the grouping matches their order in the SELECT list
+
+    For example::
+
+        SELECT department, title, AVG(salary) as avg_salary
+        FROM employees
+        GROUP BY ALL;
+
+    Is equivalent to explicitly listing all non-aggregate columns::
+
+        SELECT department, title, AVG(salary) as avg_salary
+        FROM employees
+        GROUP BY department, title;
+
+.. NOTE::
+
+  The same limitations that apply to regular GROUP BY also apply to GROUP BY ALL:
+  - Grouping can only be applied on indexed fields
+  - Non-aggregate columns in the SELECT list must appear in the grouping
+  - The HAVING clause can be used with GROUP BY ALL for filtering groups
+
+
```

Comment:
```suggestion
```

Yes I'd remove them here. If any limitations are documented they should be in the general GROUP BY section and at best linked here